parse/grammar/KuneiformParser.g4

```diff
@@ -0,0 +1,378 @@
+    (access_modifier)+
+    LBRACE action_block RBRACE
+;
+
+procedure_declaration:
+    annotation*
+    PROCEDURE IDENTIFIER
+    LPAREN (typed_variable_list)? RPAREN
+    (access_modifier)+
+    (procedure_return)?
+    LBRACE procedure_block RBRACE
+;
+
+
+foreign_procedure_declaration:
+    FOREIGN PROCEDURE IDENTIFIER
+    LPAREN (unnamed_params=type_list|named_params=typed_variable_list)? RPAREN
+    (procedure_return)?
+;
+
+procedure_return:
+    RETURNS (TABLE? LPAREN return_columns=named_type_list RPAREN
+    | LPAREN unnamed_return_types=type_list RPAREN)
+;
+
+/*
+    The following section includes parser rules for SQL.
+*/
+
+// sql is a top-level SQL statement.
+sql:
+    sql_statement SCOL
+;
+
+sql_statement:
+    (WITH common_table_expression (COMMA common_table_expression)*)?
+    (select_statement | update_statement | insert_statement | delete_statement)
+;
+
+common_table_expression:
+    IDENTIFIER LPAREN (IDENTIFIER (COMMA IDENTIFIER)*)? RPAREN AS LPAREN select_statement RPAREN
+;
+
+select_statement:
+    select_core
+    (compound_operator select_core)*
+    (ORDER BY ordering_term (COMMA ordering_term)*)?
+    (LIMIT limit=sql_expr)?
+    (OFFSET offset=sql_expr)?
+;
+
+compound_operator:
+    UNION ALL? | INTERSECT | EXCEPT
+;
+
+ordering_term:
+    sql_expr (ASC | DESC)? (NULLS (FIRST | LAST))?
+;
+
+select_core:
+    SELECT DISTINCT?
+    result_column (COMMA result_column)*
+    (FROM relation join*)?
+    (WHERE where=sql_expr)?
+    (
+        GROUP BY group_by=sql_expr_list
+        (HAVING having=sql_expr)?
+    )?
+;
+
+relation:
+    table_name=IDENTIFIER (AS? alias=IDENTIFIER)? # table_relation
+    | LPAREN select_statement RPAREN AS? alias=IDENTIFIER # subquery_relation
+    | sql_function_call AS? alias=IDENTIFIER # function_relation
+;
+
+join:
+    (INNER| LEFT | RIGHT | FULL) JOIN
+    relation ON sql_expr
+;
+
+result_column:
+    sql_expr (AS? IDENTIFIER)? # expression_result_column
+    | (table_name=IDENTIFIER PERIOD)? STAR # wildcard_result_column
+;
+
+update_statement:
+    UPDATE table_name=IDENTIFIER (AS alias=IDENTIFIER)?
+    SET update_set_clause (COMMA update_set_clause)*
+    (FROM relation join*)?
+    (WHERE where=sql_expr)?
+;
+
+update_set_clause:
+   column=IDENTIFIER EQUALS sql_expr
+;
+
+insert_statement:
+    INSERT INTO table_name=IDENTIFIER (AS alias=IDENTIFIER)?
+    (LPAREN target_columns=identifier_list RPAREN)?
+    VALUES LPAREN sql_expr_list RPAREN (COMMA LPAREN sql_expr_list RPAREN)*
+    upsert_clause?
+;
+
+upsert_clause:
+    ON CONFLICT
+    (LPAREN conflict_columns=identifier_list RPAREN (WHERE conflict_where=sql_expr)?)?
+    DO (
+        NOTHING
+        | UPDATE SET update_set_clause (COMMA update_set_clause)*
+        (WHERE update_where=sql_expr)?
+    )
+;
+
+delete_statement:
+    DELETE FROM table_name=IDENTIFIER (AS alias=IDENTIFIER)?
+    // (USING relation join*)?
+    (WHERE where=sql_expr)?
+;
+
+sql_expr:
+    literal type_cast? # literal_sql_expr
```

Comment:
It would be really clear and easier to read if the 'alternate label' are aligned on the right. I kind of have a strong feeling about it, especially when I'm reviewing it.

Reply:
cool, can do do it on all rules? That'll be super helpful